src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessModuleTests.cs

```diff
@@ -18,26 +19,30 @@ public void TestModulePropertiesExceptOnOSX()
             if (global::Interop.IsWindows)
                 fileName = string.Format("{0}.exe", CoreRunName);
 
-            // Ensure the process has loaded the modules.
-            Assert.True(SpinWait.SpinUntil(() =>
+            try
             {
-                if (_process.Modules.Count > 0)
-                    return true;
-                _process.Refresh();
-                return false;
-            }, WaitInMS));
+                // Ensure the process has loaded the modules.
+                Assert.True(SpinWait.SpinUntil(() =>
+                {
+                    if (_process.Modules.Count > 0)
+                        return true;
+                    _process.Refresh();
+                    return false;
+                }, WaitInMS));
 
-            ProcessModuleCollection moduleCollection = _process.Modules;
-            Assert.True(moduleCollection.Count > 0);
+                ProcessModuleCollection moduleCollection = _process.Modules;
+                Assert.True(moduleCollection.Count > 0);
 
-            ProcessModule coreRunModule = _process.MainModule;
-            Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
-            Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
-            Assert.Equal(fileName, coreRunModule.ModuleName);
-            Assert.True(coreRunModule.ModuleMemorySize > 0);
-            Assert.EndsWith(fileName, coreRunModule.FileName);
+                ProcessModule coreRunModule = _process.MainModule;
```

Comment:
> This may not be coreRun, right?

True, depending on the window when we check. However for most cases this will be true. I still need to follow-up with windows on this behavior and so I would suggest keep 1896 open (we did not see a lot of hits for this before and with this iteration we should be behaving much like the old tests). Let me know if you disagree.

> Should we be validating these properties for all modules, not just the first one?

Yes Please  :)